src/actions/walletActions.js

```diff
@@ -207,18 +214,41 @@ export function spendSlices(inputs, changeSlice) {
     );
 
     const updatedSlices = await Promise.all(sliceUpdates);
+    const firstSlice = updatedSlices[0];
+    const utxoCount = firstSlice.change
+      ? changeNodes[firstSlice.bip32Path].utxos.length
+      : depositNodes[firstSlice.bip32Path].utxos.length;
+
+    // want to make sure we only update the slices if they
+    // have changed states, i.e. the number utxos is different from what we have
+    // so if the slice state isn't reporting as changed after fetching
+    // we will try a few more times
+    if (firstSlice.utxos.length === utxoCount && retries)
+      return setTimeout(
+        () => dispatch(updateTxSlices(changeSlice, retries - 1)),
+        750
+      );
 
-    updatedSlices.forEach((slice) => {
+    // if we're out of retries and counts are still the same
+    // then we're done trying and should show an error
+    if (firstSlice.utxos.length === utxoCount && !retries) {
```

Comment:
If there is no change reflected, and we are out of retries, then something has gone wrong and we should report this to the user.